server/app/middlewares/websocket_backend.rb

```diff
@@ -1,9 +1,14 @@
 require 'faye/websocket'
 require_relative '../services/rpc_server'
+require_relative '../services/watchdog'
 require_relative '../services/agent/node_plugger'
 require_relative '../services/agent/node_unplugger'
 
 class WebsocketBackend
+  WATCHDOG_INTERVAL = 0.5 # seconds
+  WATCHDOG_THRESHOLD = 1.0 # seconds
+  WATCHDOG_TIMEOUT = 10.0 # seconds # XXX: testing
```

Comment:
This needs a bit of thought. If the server EM thread is overloaded, does aborting it and restarting it make things better, or worse?